File: 2609.cpp
#include <iostream>

using namespace std;

// int gcd(int x, int y) {
// 	if (y == 0) {
// 		return x;
// 	} else {
// 		return gcd(y, x%y);
// 	}
// }

int gcd(int x, int y) {
	while(y != 0) {
		int r = x%y;
		x = y;
		y = r;
	}
	return x;
}

int lcm(int x, int y) {
	int g = gcd(x, y);
	return g * x/g * y/g;
}

int main() {
	cin.tie(NULL);
	ios_base::sync_with_stdio(0);
	freopen("input.txt", "r", stdin);

	int x, y;
	cin >> x >> y;

	cout << gcd(x, y) << '\n' << lcm(x, y);

	return 0;
}

File: _2000/2609.py
import math
import sys
sys.stdin = open("input.txt", 'r')
input = sys.stdin.readline


def gcd(a, b):
    a_set = set()
    b_set = set()
    for n in range(1, int(a**0.5)+1):
        q, r = divmod(a, n)
        if r == 0:
            a_set.add(n)
            a_set.add(q)

    for n in range(1, int(b**0.5)+1):
        q, r = divmod(b, n)
        if r == 0:
            b_set.add(n)
            b_set.add(q)

    return max(a_set.intersection(b_set))


def lcm(a, b):
    return a*b//gcd(a, b)


A, B = map(int, input().split())

# print(math.gcd(A, B))  # 최대공약수
# print(math.lcm(A, B))  # 최소공배수

print(gcd(A, B))
print(lcm(A, B))
